core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/UdtCodec.java

```diff
@@ -107,10 +111,8 @@ public UdtValue decode(@Nullable ByteBuffer bytes, @NonNull ProtocolVersion prot
       int i = 0;
       while (input.hasRemaining()) {
         if (i == cqlType.getFieldTypes().size()) {
```

Comment:
This feels ok given that the only operations you can do on `ALTER TYPE` are `ADD` and `RENAME`,

Given that:

1. you can't drop a udt field;
2. adding a field is always additive (e.g. if you have existing fields, any added fields should come after, at least that is what it looks like this does: https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/statements/schema/AlterTypeStatement.java#L161-L162)

This means you won't risk reading the wrong field as position-ally everything should be retained after any `ALTER TYPE` operation.

Also considering that you can still write a udt value with the old udtCodec when the type is altered, it seems reasonable that you'd be able to read partially from it as well.

I suppose there is some inherent risk of someone creating a `UdtCodec` by hand and potentially reading the wrong interpretation of things; but I think that should probably done at your own peril anyways.

If you pull a `UdtCodec` from the registry and the type is later changed, the codec would still be usable after this change, this seems of more value than protecting against the risk I outlined.